conans/client/graph/proxy.py

```diff
@@ -62,67 +71,95 @@ def _get_recipe(self, reference, check_updates, update, remotes):
+                # TODO: cache2.0 here if we already have a revision in the cache but we add the
+                #  --update argument and we find that same revision in server, we will not
+                #  download anything but we will UPDATE the date of that revision in the
+                #  local cache and WE ARE ALSO UPDATING THE REMOTE
+                #  Check if this is the flow we want to follow
+                cache_time = self._cache.get_timestamp(ref)
+                if latest_rrev.revision != ref.revision:
+                    if cache_time < remote_time:
+                        remotes.select(remote.name)
+                        # the remote one is newer
+                        output.info("Retrieving from remote '%s'..." % remote.name)
+                        remote, new_ref = self._download_recipe(latest_rrev, output,
+                                                                remotes, remote)
+                        new_recipe_layout = self._cache.ref_layout(new_ref)
+                        new_conanfile_path = new_recipe_layout.conanfile()
+                        status = RECIPE_UPDATED
+                        return new_conanfile_path, status, remote, new_ref
+                    else:
+                        status = RECIPE_NEWER
                 else:
-                    status = RECIPE_UPDATEABLE
+                    # TODO: cache2.0 we are returning RECIPE_UPDATED just because we are updating
+                    #  the date
+                    if cache_time >= remote_time:
+                        status = RECIPE_INCACHE
+                    else:
+                        selected_remote = remote
+                        remotes.select(remote.name)
+                        self._cache.update_reference(ref,
+                                                     new_timestamp=remote_time,
+                                                     new_remote=selected_remote.name)
+                        status = RECIPE_INCACHE_DATE_UPDATED
+                return conanfile_path, status, selected_remote, ref
             else:
-                status = RECIPE_NEWER
+                status = RECIPE_NOT_IN_REMOTE
+                return conanfile_path, status, selected_remote, ref
         else:
             status = RECIPE_INCACHE
+            return conanfile_path, status, cur_remote, ref
+
+    def _get_rrev_from_remotes(self, reference, remotes, check_all_servers):
+        results = []
+        output = ScopedOutput(str(reference), self._out)
+
+        # TODO: cache2.0 --update strategies: when we have specified the revision we don't want to
+        #  check all the remotes, just return the first match
+        output.info(f"Checking all remotes: ({', '.join([remote.name for remote in remotes])})")
+
+        for remote in remotes:
+            try:
+                output.info(f"Checking remote: {remote.name}")
+                remote_rrevs = self._remote_manager.get_recipe_revisions(reference, remote)
```

Comment:
If this is a "getting latest" use case that could be implemented by the remote API, lets hide it at least behind the ``remote_manager``, instead of computing here explicitly.